From 5d36ac6bfe8c2dafb94f3220dede2ddb00fbc41e Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Tue, 14 Jun 2022 22:11:19 +0900
Subject: [PATCH] Add libchrome-specific build configs

---
 build/build_config.h | 28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/build/build_config.h b/build/build_config.h
index 5d684d154f..bdb449a57e 100644
--- a/build/build_config.h
+++ b/build/build_config.h
@@ -48,12 +48,36 @@
 
 #include "build/buildflag.h"
 
+// A brief primer on #defines:
+//
+// - __ANDROID__ is automatically defined by the Android toolchain (see
+//   https://goo.gl/v61lXa). It's not defined when building host code.
+// - __ANDROID_HOST__ is defined via -D by Android.mk when building host code
+//   within an Android checkout.
+// - ANDROID is defined via -D when building code for either Android targets or
+//   hosts. Use __ANDROID__ and __ANDROID_HOST__ instead.
+// - OS_ANDROID is a define used to build Chrome for Android within the NDK and
+//   to build Android targets.
+
+// Android targets and hosts don't use tcmalloc.
+#if defined(__ANDROID__) || defined(__ANDROID_HOST__)
+#define NO_TCMALLOC
+#endif  // defined(__ANDROID__) || defined(__ANDROID_HOST__)
+
+#if defined(__ANDROID__)  // Android targets
+
+#define OS_ANDROID 1
+
+#elif !defined(__ANDROID_HOST__)  // Chrome OS
+
+#define OS_CHROMEOS 1
+
+#endif  // defined(__ANDROID__)
+
 // A set of macros to use for platform detection.
 #if defined(__native_client__)
 // __native_client__ must be first, so that other OS_ defines are not set.
 #define OS_NACL 1
-#elif defined(ANDROID)
-#define OS_ANDROID
 #elif defined(__APPLE__)
 // Only include TargetConditionals after testing ANDROID as some Android builds
 // on the Mac have this header available and it's not needed unless the target
-- 
2.36.1.476.g0c4daa206d-goog

