diff --git a/base/hash/sha1_nacl.cc b/base/hash/sha1_nacl.cc
index 047f9ba275..2b38fdb37a 100644
--- a/base/hash/sha1_nacl.cc
+++ b/base/hash/sha1_nacl.cc
@@ -191,6 +191,7 @@ std::string SHA1HashString(StringPiece str) {
 
 void SHA1HashBytes(const unsigned char* data, size_t len, unsigned char* hash) {
   SHA1Context context;
+  context.Init();
   context.Update(data, len);
   context.Final();
 
