From ede38be7cdbe3807205a5f68cb0b69e3dfa3cd45 Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Wed, 15 Jun 2022 23:23:05 +0900
Subject: [PATCH] Remove ICU dependency for base::Time::Explode

---
 base/time/time_exploded_posix.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/base/time/time_exploded_posix.cc b/base/time/time_exploded_posix.cc
index 7fb53f7ab720..6d1d4d768711 100644
--- a/base/time/time_exploded_posix.cc
+++ b/base/time/time_exploded_posix.cc
@@ -128,12 +128,14 @@ void Time::Explode(bool is_local, Exploded* exploded) const {
 
   // For systems with a Y2038 problem, use ICU as the Explode() implementation.
   if (sizeof(SysTime) < 8) {
+#if 0 // Not to introduce icu dependency for now. Y2038 boards will be EOL by then.
 // TODO(b/167763382) Find an alternate solution for Chromecast devices, since
 // adding the icui18n dep significantly increases the binary size.
 #if !BUILDFLAG(IS_CASTOS)
     ExplodeUsingIcu(millis_since_unix_epoch, is_local, exploded);
     return;
 #endif  // !BUILDFLAG(IS_CASTOS)
+#endif
   }
 
   // Split the |millis_since_unix_epoch| into separate seconds and millisecond
-- 
2.37.2.789.g6183377224-goog

